src/components/Api.js

```diff
@@ -0,0 +1,82 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _renderResult(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfoAndCards() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  async getInitialCards() {
+    const res = await fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    });
+    return this._renderResult(res);
+  }
+
+  async addCard({ name, link }) {
+    const res = await fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({ name, link }),
+    });
+    return this._renderResult(res);
+  }
+
+  async getUserInfo() {
+    const res = await fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    });
+    return this._renderResult(res);
+  }
+
+  setUserInfo(name, about) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({ name: name, about: about }),
+    });
```

Comment:
`.then(this._renderResult);` should be used everywhere in the code to check if the response is `ok`